spec/dealership_spec.rb

```diff
@@ -0,0 +1,132 @@
+require 'rspec'
+require './lib/car'
+require './lib/dealership'
+
+describe Dealership do
+  it 'exists' do
+    dealership = Dealership.new("Acme Auto", "123 Main Street")
+
+    expect(dealership).to be_a(Dealership)
+  end
+
+  it 'has a name' do
+    dealership = Dealership.new("Acme Auto", "123 Main Street")
+
+    expect(dealership.name).to eq("Acme Auto")
+  end
+
+  it 'has an address' do
+    dealership = Dealership.new("Acme Auto", "123 Main Street")
+
+    expect(dealership.address).to eq("123 Main Street")
+  end
+
+  it 'initially has an empty array of inventory and can count inventory' do
+    dealership = Dealership.new("Acme Auto", "123 Main Street")
+    expect(dealership.inventory).to eq([])
+    expect(dealership.inventory_count).to eq(0)
+    car_1 = Car.new("Ford Mustang", 1500, 36)
+    car_2 = Car.new("Toyota Prius", 1000, 48)
+    dealership.add_car(car_1)
+    expect(dealership.inventory_count).to be(1)
+    expect(dealership.inventory).to eq([car_1])
+    dealership.add_car(car_2)
+    expect(dealership.inventory).to eq([car_1, car_2])
+    expect(dealership.inventory_count).to be(2)
+  end
+
+  it '#has_inventory?' do
+    dealership = Dealership.new("Acme Auto", "123 Main Street")
+    expect(dealership.has_inventory?).to be(false)
+    car_1 = Car.new("Ford Mustang", 1500, 36)
+    car_2 = Car.new("Toyota Prius", 1000, 48)
+    car_3 = Car.new("Toyota Tercel", 500, 48)
+    car_4 = Car.new("Chevrolet Bronco", 1250, 24)
+    dealership.add_car(car_1)
+    dealership.add_car(car_2)
+    dealership.add_car(car_3)
+    dealership.add_car(car_4)
+    expect(dealership.has_inventory?).to be(true)
+  end
+
+  it 'can tell you the vehicles by selected make in inventory' do
+    dealership = Dealership.new("Acme Auto", "123 Main Street")
+    car_1 = Car.new("Ford Mustang", 1500, 36)
+    car_2 = Car.new("Toyota Prius", 1000, 48)
+    car_3 = Car.new("Toyota Tercel", 500, 48)
+    car_4 = Car.new("Chevrolet Bronco", 1250, 24)
+    dealership.add_car(car_1)
+    dealership.add_car(car_2)
+    dealership.add_car(car_3)
+    dealership.add_car(car_4)
+    expect(dealership.cars_by_make("Toyota")).to eq([car_2, car_3])
+    expect(dealership.cars_by_make("Ford")).to eq([car_1])
+  end
+
+  it 'can tell you the total value of cars in the dealership' do
+    dealership = Dealership.new("Acme Auto", "123 Main Street")
+    car_1 = Car.new("Ford Mustang", 1500, 36)
+    car_2 = Car.new("Toyota Prius", 1000, 48)
+    car_3 = Car.new("Toyota Tercel", 500, 48)
+    car_4 = Car.new("Chevrolet Bronco", 1250, 24)
```

Comment:
Can you think of a way to use `before(:each)` or `let` so that you don't need to redefine the dealership and cars for every test?